hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -321,6 +321,19 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		return ctrl.Result{}, fmt.Errorf("failed to look up release image metadata: %w", err)
 	}
 
+	// Validate the HostedCluster payload can support the NodePool's CPU arch
+	if err = validateHCPayloadSupportsNodePoolCPUArch(hcluster, nodePool); err != nil {
```

Comment:
Also this check needs to be grouped with one below for 	if !isArchAndPlatformSupported(nodePool) {

Reply:
> This would temporarily short circuit every NodePool across the fleet until the field is set.
Wouldn't it only affect the particular NodePool that is reconciling here? 

Updated to set a status condition on success.

Reply:
>Wouldn't it only affect the particular NodePool that is reconciling here?

When you upgrade the image in a Deployment. It does a rolling upgrade of the Pods. The pod start running a the new controller with new code. As the controller starts it reconcile all existing CRs is watching.